temba/api/v2/views.py

```diff
@@ -3573,3 +3581,454 @@ def post(self, request, *args, **kwargs):
+            "X-Parse-Master-Key": settings.PARSE_MASTER_KEY,
+            "Content-Type": "application/json",
+        }
+        results_url = f"{settings.PARSE_URL}/classes/{collection}?order=order&limit=1000"
+        response = requests.get(results_url, headers=parse_headers)
+        result = response.json()
+
+        return Response(result, status=status.HTTP_200_OK)
+
+    def post(self, request, *args, **kwargs):
+        org, collection_name, collections_list, error_response = self.get_default_params(config="collection_exist")
+        if error_response:
+            return error_response
+
+        items_to_push = self.request.data.get("items", [])
+        if not items_to_push:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        collection = self.get_collection_full_name(org=org, collection=collection_name)
+        count_url = f"{settings.PARSE_URL}/classes/{collection}?count=1"
+        count_response = requests.get(count_url, headers=self.parse_headers)
+
+        if count_response.status_code == 200:
+            requests_ = []
+            insert_url = f"{settings.PARSE_URL}/batch"
+            insert_index = count_response.json().get("count")
+            db_endpoint = f"{settings.PARSE_ENDPOINT}/classes/{collection}"
+            for index, data in enumerate(items_to_push, start=insert_index):
+                requests_.append({"method": "POST", "path": db_endpoint, "body": {"order": index, **data}})
+            response = requests.post(insert_url, data=json.dumps({"requests": requests_}), headers=self.parse_headers)
+        else:
+            return Response(count_response.json(), status=count_response.status_code)
+
+        return Response(
+            response.json(), status=(status.HTTP_201_CREATED if response.status_code == 200 else response.status_code)
+        )
+
+    def delete(self, request, *args, **kwargs):
+        org, collection_name, collections_list, error_response = self.get_default_params(config="collection_exist")
+        if error_response:
+            return error_response
+
+        object_id = self.request.data.get("objectId")
+        if not object_id:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        collection = self.get_collection_full_name(org=org, collection=collection_name)
+        parse_url = f"{settings.PARSE_URL}/classes/{collection}/{object_id}"
+        response = requests.delete(parse_url, headers=self.parse_headers)
+
+        return Response(status=(status.HTTP_204_NO_CONTENT if response.status_code == 200 else response.status_code))
+
+    def put(self, request, *args, **kwargs):
+        org, collection_name, collections_list, error_response = self.get_default_params(config="collection_exist")
+        if error_response:
+            return error_response
+
+        object_id = self.request.data.get("objectId")
+        if not object_id:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Let's add a message to this response, this way the user would know what's wrong.

Reply:
Added response details.